src/unix/linux-core.c

```diff
@@ -209,8 +218,15 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
   assert(timeout >= -1);
   base = loop->time;
   count = 48; /* Benchmarks suggest this gives the best throughput. */
+  real_timeout = timeout;
 
   for (;;) {
+    /* See the comment for max_safe_timeout for an explanation of why
+     * this is necessary.  Executive summary: kernel bug workaround.
+     */
+    if (sizeof(int32_t) == sizeof(long) && timeout >= max_safe_timeout)
```

Comment:
I'm not sure I understand the question.  A timeout < 0 means "poll indefinitely", zero means "don't block" and > 0 means "block for that many milliseconds or until a file descriptor becomes ready."

In theory, epoll_wait() should be able to block for 2**31-1 milliseconds or about 24 days.  But because of a kernel bug, any timeout > 30 minutes turns into an indefinite wait.

In libuv, it only affects timers, and only when there isn't anything else happening.